### PR TITLE
DNC Technical Finish Windows

### PR DESCRIPTION
**Pull request by AkairyuGestalter**
Tracking Technical Finish stuff as a buff window
Devilment suggestions here now
Groundwork for letting the player know they should/shouldn't have let RW drop
Adjusted the procs suggestion wording in the meantime
Also set up a defined display order for the modules

### CODE REVIEW

File: src/parser/jobs/dnc/modules/DISPLAY_ORDER.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export default {
```
Comment (ackwell) [inline src/parser/jobs/dnc/modules/DISPLAY_ORDER.ts]: technically not a module, makes more sense in the parent dir

File: src/parser/jobs/dnc/modules/DirtyDancing.tsx [hunk 2]
```diff
@@ -354,7 +327,7 @@ export default class DirtyDancing extends Module {
 		}
 	}
 
-	private getNotesIcon(ruleFailed: boolean): any {
+	private getNotesIcon(ruleFailed: boolean): TODO {
```
Comment (ackwell) [inline src/parser/jobs/dnc/modules/DirtyDancing.tsx]: Should be able to drop the return type annotation - pretty sure it'll be a `JSX.Element`

File: src/parser/jobs/dnc/modules/Technicalities.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+import Module, {dependency} from 'parser/core/Module'
+import Combatants from 'parser/core/modules/Combatants'
+import {AoeEvent} from 'parser/core/modules/Combos'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import Timeline from 'parser/core/modules/Timeline'
+
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+import FeatherGauge from './FeatherGauge'
+
+// Harsher than the default since you'll only have 4-5 total windows anyways
```
Comment (ackwell) [inline src/parser/jobs/dnc/modules/Technicalities.tsx]: Can skip straight to medium if you think it's warranted

File: src/parser/jobs/dnc/modules/Technicalities.tsx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+	timelyDevilment: boolean = true
+	poolingProblem: boolean = false
+
+	constructor(start: number) {
+		this.start = start
+	}
+}
+
+export default class Technicalities extends Module {
+	static handle = 'technicalities'
```
Comment (ackwell) [inline src/parser/jobs/dnc/modules/Technicalities.tsx]: quite an apt name given how much fiddling around this has taken :meowlul:

File: src/parser/jobs/dnc/modules/Technicalities.tsx [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+			return
+		}
+
+		if (this.isWindowOkToClose(lastWindow, event)) {
+			lastWindow.end = event.timestamp
+
+			// Check to see if this window could've had more feathers due to possible pooling problems
+			if (this.feathers.feathersSpentInRange(lastWindow.start, lastWindow.end) < FEATHER_THRESHHOLD) {
+				const previousWindow = this.history[this.history.length-2]
+				const feathersBeforeWindow = this.feathers.feathersSpentInRange((previousWindow ? previousWindow.start : this.parser.fight.start_time)
```
Comment (ackwell) [inline src/parser/jobs/dnc/modules/Technicalities.tsx]: Should this be `previousWindow.end`? potentially going to pick up spent feathers in the previous window's duration as-is, i _think_.

Comment (AkairyuGestalter) [reply src/parser/jobs/dnc/modules/Technicalities.tsx]: Hurrr, yes, you're right. Don't code while MSQlette-ing, kids. I'll fix that tonight.

File: src/parser/jobs/dnc/modules/Technicalities.tsx (hunk 3 above)
Comment (AkairyuGestalter) [reply src/parser/jobs/dnc/modules/Technicalities.tsx]: Still wanted to leave the possibility of a herp or fat-finger or something and not be too mean about it. They get one slip-up though, after that it's Med+

**Pull request merged**